src/hotspot/share/runtime/synchronizer.cpp

```diff
@@ -607,7 +606,7 @@ void ObjectSynchronizer::jni_exit(oop obj, TRAPS) {
   // If this thread has locked the object, exit the monitor. We
   // intentionally do not use CHECK on check_owner because we must exit the
   // monitor even if an exception was already pending.
-  if (monitor->check_owner(THREAD)) {
+  if (monitor->check_owner(current)) {
```

Comment:
We can actually throw IMSE from check_owner() if this thread is not the real owner.

Reply:
I reverted this change.  Thanks.